public/app/plugins/datasource/prometheus/components/PromQueryEditor.tsx

```diff
@@ -24,11 +24,20 @@ const INTERVAL_FACTOR_OPTIONS: Array<SelectableValue<number>> = map([1, 2, 3, 4,
   label: '1/' + value,
 }));
 
+const STEP_OPTIONS: Array<SelectableValue<number>> = map([0, 1, 2], (value: number) => {
```

Comment:
I think you can just inline this, with 3 options mapping over an array seems to add more boilerplate than it removes.

Also you can have a `type StepType = 'min' | 'max' | 'exact'` and that will allow only those 3 strings (you could also do an enum but this string type is probably enough). Then you would have `SelectableValue<StepType>` and you would get some type safety.

Also later on in `adjustInterval` you could safely check `stepOption === 'min'` and I think compiler should complain if you misspell that string.

Reply:
I can't seem to get it to work though. The `options` prop of the Select component requires `SelectableValue<number>`

Reply:
Never mind, i solved it!